Driverless AI/scoring-pipeline-deployment-to-apache-nifi/scoring-pipeline-deployment-to-apache-nifi.md

```diff
@@ -0,0 +1,352 @@
+
+## Outline
+
+- [Objective](#objective)
+- [Prerequisites](#prerequisites)
+- [Task 1: Set Up Environment](#task-1-set-up-environment)
+- [Task 2: NiFi and MOJO Scoring Pipeline Concepts](#task-2-nifi-and-mojo-scoring-pipeline-concepts)
+- [Task 3: Batch Scoring](#task-3-batch-scoring)
+- [Task 4: Real-Time Scoring](#task-4-real-time-scoring)
+- [Task 5: Challenge](#task-5-challenge)
+- [Next Steps](#next-steps)
+- [Appendix A: Glossary](#appendix-a-glossary)
+
+## Objective
+**Machine Learning Model Deployment** is the process of making your model available in production environments, so they can be used to make predictions for other software systems [1]. Before model deployment, **feature engineering** occurs in the form of preparing data that later on will be used to train a model [2]. Driverless AI **Automatic Machine Learning (AutoML)** combines the best feature engineering and one or more **machine learning models** into a scoring pipeline [3][4]. The scoring pipeline is used to score or predict data when given new test data [5]. The **scoring pipeline** comes in two flavors. The first scoring pipeline is a **Model Object, Optimized (MOJO) Scoring Pipeline**, which is a standalone, low-latency model object designed to be easily embeddable in production environments. The second scoring pipeline is a Python Scoring Pipeline, which has a heavy footprint that is all Python and uses the latest libraries of Driverless AI to allow for executing custom scoring recipes[6].
+
+By the end of this tutorial, you will predict the **cooling condition** for a Hydraulic System Test Rig by deploying an **embeddable MOJO Scoring Pipeline** to **Apache NiFi**. The Hydraulic System Test Rig data comes from [UCI Machine Learning Repository: Condition Monitoring of Hydraulic Systems Data Set](https://archive.ics.uci.edu/ml/datasets/Condition+monitoring+of+hydraulic+systems#). Hydraulic System Test Rigs are used to test components in Aircraft Equipment, Ministry of Defense, Automotive Applications, and more [7]. This Hydraulic Test Rig is capable of testing a range of flow rates that can achieve different pressures with the ability to heat and cool to simulate testing under different conditions [8]. Testing the pressure, volume flow and temperature is possible by Hydraulic Test Rig sensors and digital displays. The display panel alerts the user when certain testing criteria is met displaying either a green/red light [8]. A filter blockage panel indicator is integrated into the panel to ensure the Hydraulic Test Rig’s oil is maintained [8]. The cooling filtration solution is designed to minimize power consumption and expand the life of the Hydraulic Test Rig. We are predicting cooling conditions for Hydraulic System Predictive Maintenance. When the cooling condition is low, our prediction tells us that the cooling of the Hydraulic System is close to total failure and we may need to look into replacing the cooling filtration solution soon.
```

Comment:
add "the" after this "The Hydraulic System Test Rig data comes from"

Reply:
rewrite this sentence: "Testing the pressure, volume flow and temperature is possible by Hydraulic Test Rig sensors and digital displays."

to this: "Testing the pressure, volume flow, and the temperature is possible by Hydraulic Test Rig sensors and digital displays."